packages/backend/src/apps/m365-excel/dynamic-data/list-table-columns/index.ts

```diff
@@ -0,0 +1,37 @@
+import type { IDynamicData } from '@plumber/types'
+
+import { getM365TenantInfo } from '@/config/app-env-vars/m365'
+
+import { getRegisteredAuthData } from '../../common/auth-data'
+
+const dynamicData: IDynamicData = {
+  name: 'List Table Columns',
+  key: 'listTableColumns',
+
+  async run($) {
+    const { fileId, tableId } = $.step.parameters
+    if (!fileId || !tableId) {
+      return {
+        data: [],
+      }
+    }
+
+    const authData = getRegisteredAuthData($)
+    const tenant = getM365TenantInfo(authData.tenantKey)
+    const results = await $.http.get<{
+      value: Array<{ index: number; name: string }>
+    }>(
+      `/v1.0/sites/${tenant.sharePointSiteId}/drive/items/${fileId}/workbook/tables/${tableId}/columns?$select=index,name&$orderby=index`,
+    )
+
+    return {
+      data: results.data.value.map((entry) => ({
+        name: `Column ${entry.index} (${entry.name})`,
+        // We return indices because excel works with indices, not names.
+        value: String(entry.index),
+      })),
```

Comment:
this might be quite confusing to users

<img width="776" alt="image" src="https://github.com/opengovsg/plumber/assets/10072985/0fff6d69-4c4d-4932-9237-cd16e125e835">

suggest 
name: entry.name,
value: `Column ${entry.index + 1}`

and then parse the value back to index later on

Reply:
Thanks I was mentioning to stacey how this screenshot would give her heart attack

Reply:
Did a little more digging and I discovered that I can match column name to its index without needing an additional query. Coded that up instead, seems more intuitive.